test/unit/Cardano/Wallet/PrimitiveSpec.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.PrimitiveSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Primitive
+    ( Address (..)
+    , Coin (..)
+    , Dom (..)
+    , Hash (..)
+    , TxIn (..)
+    , TxOut (..)
+    , UTxO (..)
+    , excluding
+    , isSubsetOf
+    , restrictedBy
+    , restrictedTo
+    )
+import Data.Set
+    ( Set, (\\) )
+import Test.Hspec
+    ( Spec, describe, it )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , Property
+    , checkCoverage
+    , choose
+    , cover
+    , property
+    , scale
+    , vectorOf
+    , (===)
+    )
+
+import qualified Data.ByteString as BS
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+
+
+spec :: Spec
+spec = do
+    describe "Lemma 2.1 - Properties of UTxO operations" $ do
+        it "2.1.1) ins⊲ u ⊆ u" (checkCoverage prop_2_1_1)
+        it "2.1.2) ins⋪ u ⊆ u" (checkCoverage prop_2_1_2)
+        it "2.1.3) u ⊳ outs ⊆ u" (checkCoverage prop_2_1_3)
+        it "2.1.4) ins⊲ (u ⋃ v) = (ins⊲ u) ⋃ (ins⊲ v)" (checkCoverage prop_2_1_4)
+        it "2.1.5) ins⋪ (u ⋃ v) = (ins⋪ u) ⋃ (ins⋪ v)" (checkCoverage prop_2_1_5)
+        it "2.1.6) (dom u ⋂ ins) ⊲ u = ins⊲ u" (checkCoverage prop_2_1_6)
+        it "2.1.7) (dom u ⋂ ins) ⋪ u = ins⋪ u" (checkCoverage prop_2_1_7)
+        it "2.1.8) (dom u ⋃ ins) ⋪ (u ⋃ v) = (ins ⋃ dom u) ⋪ v" $ checkCoverage prop_2_1_8
+        it "2.1.9) ins⋪ u = (dom u \\ ins)⊲ u" (checkCoverage prop_2_1_9)
+
+{-------------------------------------------------------------------------------
+       Wallet Specification - Lemma 2.1 - Properties of UTxO operations
+-------------------------------------------------------------------------------}
+
+prop_2_1_1 :: (Set TxIn, UTxO) -> Property
+prop_2_1_1 (ins, u) =
+    cover 50 cond "dom u ⋂ ins ≠ ∅" (property prop)
```

Comment:
Though the condition and coverage threshold isn't the same, albeit identical for several. We could go and have a couple of general helpers but I think it would just mess up readability in the end, and props are already rather self contained :/ ...